tests/post/steps/test_example_solution.py

```diff
@@ -0,0 +1,398 @@
+SOLUTION_CONFIGURATION_FILE = "/etc/metalk8s/solutions.yaml"
+INFRA_TAINT = "node-role.kubernetes.io/infra:NoSchedule-"
+BOOTSTRAP_TAINT = "node-role.kubernetes.io/bootstrap:NoSchedule-"
+SOLUTION_CONFIGMAP = "metalk8s-solutions"
+SOLUTION_NAMESPACE = "metalk8s-solutions"
+
+# }}}
+
+# Fixtures {{{
+
+@pytest.fixture
+def k8s_appsv1_client(k8s_apiclient):
+    return AppsV1Api(api_client=k8s_apiclient)
+
+
+@pytest.fixture(scope='function')
+def context():
+    return {}
+
+
+# }}}
+
+
+# Scenarios {{{
+
+@scenario('../features/example_solution.feature', 'Deploy Solution')
+def test_deploy_solution(host):
+    pass
+
+# }}}
+
+
+# Given {{{
+
+@given(parsers.parse("no Solution '{name}' is imported"))
+def is_absent_solution(host, name, k8s_client):
+    with host.sudo():
+        if name in host.mount_point.get_mountpoints():
+            assert False, 'Found Solution mountpoint {}'.format(name)
+    return name
+
+
+@given(parsers.parse("no Solution environment '{environment}' is available"))
+def is_absent_solution_environment(host, environment, k8s_client):
+    solution_environment = get_environment(k8s_client, environment)
+    if solution_environment is not None:
+        pytest.fail("Found Solution environment {}".format(environment))
+    return environment
+
+
+@given(parsers.parse("the Solution Configuration file is absent"))
+def is_absent_solution_config(host):
+    with host.sudo():
+        if host.file(SOLUTION_CONFIGURATION_FILE).exists:
+            assert False, 'Solution Config file exist at path {}'.format(
+                        SOLUTION_CONFIGURATION_FILE
+                    )
+
+# }}}
+
+
+# When {{{
+
+@when(parsers.parse("we import a Solution archive '{path}'"))
+def import_solution(request, host, path):
+    iso_root = request.config.getoption("--iso-root")
+    cmd = str(iso_root / "solutions.sh")
+    with host.sudo():
+        res = host.run(' '.join([
+            cmd,
+            'import --archive',
+            '{}'.format(path),
+        ]))
+        assert res.rc == 0, res.stdout
+
+
+@when(parsers.parse("I activate Solution '{name}' with version '{version}'"))
+def activate_example_solution(request, host, name, version):
+    iso_root = request.config.getoption("--iso-root")
+    cmd = str(iso_root / "solutions.sh")
+    with host.sudo():
+        res = host.run(' '.join([
+            cmd,
+            'activate --name {}'.format(name),
+            '--version {}'.format(version),
+        ]))
+        assert res.rc == 0, res.stdout
+
+
+@when(parsers.parse(
+    "I create a solution environment '{environment}' for Solution '{name}'"))
+def create_solution_environment(request, host, environment, name):
+    iso_root = request.config.getoption("--iso-root")
+    cmd = str(iso_root / "solutions.sh")
+    with host.sudo():
+        res = host.run(' '.join([
+            cmd,
+            'create-env --name {}'.format(environment),
+        ]))
+        assert res.rc == 0, res.stdout
```

Comment:
Same here, code duplication, you could have an helper for this, with something like:
```
def run_solutions_command(request, host, args):
    iso_root = request.config.getoption("--iso-root")
    cmd = [str(iso_root / "solutions.sh")] + args
    with host.sudo():
        res = host.run(' '.join(cmd))
        assert res.rc == 0, res.stdout


@when(parsers.parse(
    "I create a solution environment '{environment}' for Solution '{name}'"))
def create_solution_environment(request, host, environment, name):
    run_solutions_command(request, host, ["create-env", "--name", environment])
```
And the `{name}` parameter is never used.